test/battle/ability/poison_heal.c

```diff
@@ -0,0 +1,62 @@
+#include "global.h"
+#include "test/battle.h"
+
+SINGLE_BATTLE_TEST("Poison Heal heals from Poison damage")
+{
+    GIVEN {
+        PLAYER(SPECIES_SHROOMISH) { Ability(ABILITY_POISON_HEAL); Status1(STATUS1_POISON);  HP(1), MaxHP(400); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_CELEBRATE); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_POISON_HEAL);
+        MESSAGE("The poisoning healed Shroomish a little bit!");
+        HP_BAR(player, damage: -50);
+    }
+}
+
+SINGLE_BATTLE_TEST("Poison Heal heals from Toxic Poison damage")
```

Comment:
This test should account for the fact that Toxic Poison heals at the same rate as regular Poison.